lib/auth/dbobjectimportrule/dbobjectimportrulev1/service.go

```diff
@@ -0,0 +1,194 @@
+	}
+	out, err := bs.backend.GetDatabaseObjectImportRule(ctx, req.Name)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return out, nil
+}
+
+// ListDatabaseObjectImportRules lists all DatabaseObjectImportRules.
+func (bs *DatabaseObjectImportRuleService) ListDatabaseObjectImportRules(
+	ctx context.Context, req *pb.ListDatabaseObjectImportRulesRequest,
+) (*pb.ListDatabaseObjectImportRulesResponse, error) {
+	_, err := authz.AuthorizeWithVerbs(
+		ctx, bs.logger, bs.authorizer, false, types.KindDatabaseObjectImportRule, types.VerbList,
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	out, next, err := bs.backend.ListDatabaseObjectImportRules(ctx, int(req.PageSize), req.PageToken)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &pb.ListDatabaseObjectImportRulesResponse{
+		Rules:         out,
+		NextPageToken: next,
+	}, nil
+}
+
+// CreateDatabaseObjectImportRule creates a new DatabaseObjectImportRule. It will throw an error if the DatabaseObjectImportRule already
+// exists.
+func (bs *DatabaseObjectImportRuleService) CreateDatabaseObjectImportRule(
+	ctx context.Context, req *pb.CreateDatabaseObjectImportRuleRequest,
+) (*pb.DatabaseObjectImportRule, error) {
+	_, err := authz.AuthorizeWithVerbs(
+		ctx, bs.logger, bs.authorizer, false, types.KindDatabaseObjectImportRule, types.VerbCreate,
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := databaseobjectimportrule.ValidateDatabaseObjectImportRule(req.Rule); err != nil {
+		return nil, trace.Wrap(err, "validating rule")
+	}
+	out, err := bs.backend.CreateDatabaseObjectImportRule(ctx, req.Rule)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return out, nil
+
+}
+
+// UpsertDatabaseObjectImportRule creates a new DatabaseObjectImportRule or forcefully updates an existing DatabaseObjectImportRule.
+// This is a function rather than a method so that it can be used by the gRPC service
+// and the auth server init code when dealing with resources to be applied at startup.
+func UpsertDatabaseObjectImportRule(
+	ctx context.Context,
+	backend Backend,
+	rule *pb.DatabaseObjectImportRule,
+) error {
+	if err := databaseobjectimportrule.ValidateDatabaseObjectImportRule(rule); err != nil {
+		return trace.Wrap(err, "validating rule")
+	}
+	return trace.Wrap(backend.UpsertDatabaseObjectImportRule(ctx, rule), "upserting rule")
+}
+
+// UpdateDatabaseObjectImportRule updates an existing DatabaseObjectImportRule. It will throw an error if the DatabaseObjectImportRule does
+// not exist.
+func (bs *DatabaseObjectImportRuleService) UpdateDatabaseObjectImportRule(
+	ctx context.Context, req *pb.UpdateDatabaseObjectImportRuleRequest,
+) (*pb.DatabaseObjectImportRule, error) {
+	authCtx, err := authz.AuthorizeWithVerbs(
+		ctx, bs.logger, bs.authorizer, false, types.KindDatabaseObjectImportRule, types.VerbUpdate,
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := authz.AuthorizeAdminAction(ctx, authCtx); err != nil {
```

Comment:
Why don't we perform this call in all of the other methods?

Reply:
We do now - for those that are in scope.